docs/adr/0002-schema-first-api-design.md

```diff
@@ -0,0 +1,59 @@
+# ADR 0002: Schema-First API Design
+
+## Status
+Voorstel
+
+## Context
+Om op een consistente en duidelijke manier een goed onderhoudbare contract voor de BRP API te kunnen ontwerpen en te kunnen documenteren, is er behoefte aan een werkwijze die dit faciliteert.
+
+## Beslissing
+We hebben besloten om het contract voor de BRP API te documenteren met behulp van de [OpenAPI Specificatie][OAS] en de [Schema-First API Design][SchemaFirst] aanpak te hanteren voor de BRP API ontwikkeling.
+Dit betekent dat voor elk te implementeren wijziging/toevoeging eerst wordt gekeken of de implementatie wijziging/toevoeging zal leiden tot een wijziging van het API contract. De wijziging aan het API contract wordt dan vóór de implementatie wijziging/toevoeging uitgevoerd.
+
+## Consequenties
+- **Voordelen:**
+    - Duidelijke en consistente API contract.
+    - Verbeterde communicatie tussen de stakeholders.
+    - Eerder en gemakkelijker valideren van het API contract.
+    - Betere Developer Experience (betere documentatie, client code generatie en consumers en provider van het API contract kunnen parallel worden geïmplementeerd).
+
+- **Nadelen:**
+    - Mogelijke leercurve voor stakeholders die niet bekend zijn met Schema-First API design.
+
```

Comment:
Een ander mogelijk nadeel (zien we bij gezag) is dat je geen controlle hebt over de gegenereerde code. De keuze die gemaakt zijn door de code generatie library die je gebruikt, moet je het met doen en die code is niet altijd optimaal

Reply:
opgenomen in het document